### PR TITLE
add extra guard to fail if trying to submit a too big transaction

### PR DESCRIPTION
**Pull request by KtorZ**
## Description

This can happen because we only _estimate_ the maximum number of inputs allowed for the coin selection.
As a result, the transaction generated as a result of the coin selection may be slightly bigger than
the maximum sized allowed by the network (currently 8kb). So, having an extra sanity checks prevent
wrong transactions to be submitted (and added to our pendin set)

## Linked issue

?

## Type of change

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [ ] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [ ] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [ ] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## How to merge

Send the message `bors r+` to merge this PR. For more information, see
[`docs/how-to/bors.md`](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/how-to/bors.md).

### CODE REVIEW

File: wallet/src/Cardano/Wallet/Kernel/Transactions.hs [hunk 1]
```diff
@@ -173,19 +175,30 @@ pay activeWallet spendingPassword opts accountId payees = do
...
-                          retriesLeft <- applyPolicy retryPolicy rs
-                          return . Left $ case retriesLeft of
-                               Nothing ->
-                                   PaymentSubmissionMaxAttemptsReached
-                               Just _  ->
-                                   PaymentNewPendingError e
-                      Right meta -> return $ Right (taTx $ txAux, meta)
+                 let sz = fromIntegral $ BL.length $ serialize txAux
+                 maxSz <- Node.getMaxTxSize (walletPassive activeWallet ^. walletNode)
+                 if sz > maxSz then
```
Comment (piotr-iohk) [inline wallet/src/Cardano/Wallet/Kernel/Transactions.hs]: @KtorZ, just to make sure, we don't need `sz >= maxSz` here (most probably not, but asking just to be safe :) )

Comment (KtorZ) [reply wallet/src/Cardano/Wallet/Kernel/Transactions.hs]: I am not sure how cardano-sl approaches maximum actually and whether it treats it as an inclusive or exclusive boundary :thinking: ...

File: wallet/src/Cardano/Wallet/Kernel/Transactions.hs [hunk 2]
```diff
@@ -177,7 +177,7 @@ pay activeWallet spendingPassword opts accountId payees = do
              Right (txAux, partialMeta, _utxo) -> do
                  let sz = fromIntegral $ BL.length $ serialize txAux
                  maxSz <- Node.getMaxTxSize (walletPassive activeWallet ^. walletNode)
-                 if sz > maxSz then
+                 if sz >= maxSz then
```
Comment (KtorZ) [inline wallet/src/Cardano/Wallet/Kernel/Transactions.hs]: :ok_hand:

**Comment by disassembler** [comment]
@piotr-iohk can you approve the PR so I can merge it?

**Pull request merged**